librato_python_web/instrumentor/web/cherrypy_.py

```diff
@@ -39,55 +39,51 @@
 logger = getCustomLogger(__name__)
 
 
-def _cherrypy_respond_wrapper(f):
-    def decorator(*args, **keywords):
-        try:
-            telemetry.count('web.requests')
-            Timing.push_timer()
+def _cherrypy_respond_wrapper(func, *args, **keywords):
+    try:
+        telemetry.count('web.requests')
+        Timing.push_timer()
 
-            # call the request function
-            response = f(*args, **keywords)
+        # call the request function
+        response = func(*args, **keywords)
 
-            if response.status:
-                telemetry.count('web.status.%sxx' % response.status[0:1])
-            return response
-        except Exception as e:
-            telemetry.count('web.errors')
-            raise e
-        finally:
-            try:
-                elapsed, net_elapsed = Timing.pop_timer()
-                telemetry.record('web.response.latency', elapsed)
-                telemetry.record('app.response.latency', net_elapsed)
-            except:
-                logger.exception('Teardown handler failed')
-                raise
-    return decorator
+        if response.status:
+            telemetry.count('web.status.%sxx' % response.status[0:1])
+        return response
+    except Exception as e:
+        telemetry.count('web.errors')
+        raise e
+    finally:
+        try:
+            elapsed, net_elapsed = Timing.pop_timer()
+            telemetry.record('web.response.latency', elapsed)
+            telemetry.record('app.response.latency', net_elapsed)
+        except:
+            logger.exception('Teardown handler failed')
+            raise
 
 
-def _cherrypy_wsgi_call(f):
-    def decorator(*args, **keywords):
-        t = time.time()
-        try:
-            return f(*args, **keywords)
-        finally:
-            elapsed = time.time() - t
-            telemetry.record('wsgi.response.latency', elapsed)
-    return decorator
+def _cherrypy_wsgi_call(func, *args, **keywords):
+    t = time.time()
+    try:
+        return func(*args, **keywords)
+    finally:
+        elapsed = time.time() - t
+        telemetry.record('wsgi.response.latency', elapsed)
```

Comment:
This is a bit vexing because the patterns are similar, yet different due to the difference in the web frameworks. I'll table a more careful refactoring for later.